### ISSUE TITLE
Need for function similar to torch.no_grad and torch.no_inference_mode?

### ISSUE THREAD

**Issue by Viditagarwal7479** [ml-explore/mlx#272]
I can use `mx.grad` and `mx.value_and_grad` for calculating the gradients of an `mx.array`. Since, mlx is lazy and both these functions will calculate the value and grad on the spot here we might not need such an alternative.
So, is there any need for function which will help in saving memory and compute when we don't need to compute gradient?

**Comment by Viditagarwal7479** [comment]
For models the base class `mlx.nn.Module` provides the feature to freeze certain parameters. https://github.com/ml-explore/mlx/blob/f91f45014199e35a2c4578ced2a5350890ee1368/python/mlx/nn/layers/base.py#L21

**Comment by awni** [comment]
I'm not following your request, maybe an example would help? If you want inference with no gradients you can just do:

`model(inputs)`

No temporary state is saved for backward, so it should require a lot less memory.

Feel free to comment and I will reopen if I am missing something.

**Comment by Viditagarwal7479** [comment]
Thanks, the primary query was regarding the model inference part itself.
But for example in the model initialization functions like `kaiming uniform` and `kaiming normal` we need to ensure that the operations performed while initialization aren't considered while computing gradients.

**Comment by awni** [comment]
You shouldn't run into that as an issue: `mx.grad(fun)` only takes gradients with respect to its inputs which in the common setup are the model weights after initialization...

**Comment by Viditagarwal7479** [comment]
Ah, Thank you ✨
